plugins/content_distribution/providers/cortex/lib/CortexApiDistributionEngine.php

```diff
@@ -467,6 +467,16 @@ protected function uploadVideo(KalturaCortexApiDistributionJobProviderData $apiD
 		}
 		catch(Exception $e)
 		{
+			if ($needDel)
```

Comment:
@amiras89 You should add :
`if ($needDel && kFile::checkFileExists($videoFilePath))`
 If an exception been thrown in line 430, the unlink will be failed because there is still no file here

Reply:
IF the file is shared, and failed to unlink, it will return false.
If it's local and failed, we suppress the php notice.

If exception is thrown at 430, $needDel is false - so we won't try to unlink

Because $needDel is only set to true at 447 when we copy from s3 to the .dme file - it's sufficient IMHO
Only problem is that if $videoFilePathNew is an existing .dme - then $needDel won't be true and we won't delete
But for that we need a re-design in the flow, so need to better understand the dme as we talked over in Teams

I wanted to only check if file exists --> if yes --> delete
But then we might end up deleting the file from s3, so stayed with checking the $needDel
Better to be safe than sorry.